crates/sargon-uniffi/src/profile/mfa/security_structures/builder.rs

```diff
@@ -0,0 +1,583 @@
+#![allow(clippy::new_without_default)]
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use std::{
+    borrow::Borrow,
+    sync::{Arc, RwLock},
+};
+
+use sargon::{IndexSet, MatrixBuilder};
+
+use crate::prelude::*;
+
+#[derive(Debug, uniffi::Object)]
+pub struct SecurityShieldBuilder {
+    wrapped: RwLock<Option<MatrixBuilder>>,
+    name: RwLock<String>,
+}
```

Comment:
hmm not sure, maybe! The primary work done here is that I've added `name` part o state and then juggling of the `RwLock`.... If I were to make a `SecurityShieldBuilder` in `sargon`  crate it will be quite thin, and we still need to juggle the RwLock here in `sargon-uniffi` crate. Hmm.

Reply:
@GhenadieVP ok I will add a `SecurityShieldBuilder` in `sargon` which maps all `NotYetValid` -> into `Ok()` and just `BasicViolation` or `ForeverInvalid` -> `Err`. 

And then I will make this type in `sargon-uniffi` even more stupid, sound good?